### PR TITLE
MESMER-X Linting

### PR DESCRIPTION
**Pull request by veni-vidi-vici-dormivi**

### CODE REVIEW

File: mesmer/mesmer_x/OLD_train_l_distrib.py [hunk 1]
```diff
@@ -1566,7 +1566,7 @@ def _test_evol_params(self, data_fit, args):
                 np.isclose(0, scale + c * (loc - data_fit))
             ):  # rewritten for simplicity as scale + c * (loc - data) > 0
                 test = False
-            # if type(c) == np.float64:## no covariants on shape
+            # if type(c) == np.float64:# no covariants on shape
```
Comment (mathause) [inline mesmer/mesmer_x/OLD_train_l_distrib.py:1569]:
```suggestion
            # if type(c) == np.float64: # no covariants on shape
```

File: mesmer/mesmer_x/temporary_spatial.py
Comment (mathause) [inline mesmer/mesmer_x/temporary_spatial.py:1]: It should be possible to delete this file and update the imports.

File: mesmer/mesmer_x/train_l_distrib_mesmerx.py [hunk 2]
```diff
@@ -15,8 +15,15 @@
 from scipy.interpolate import RegularGridInterpolator
 from scipy.optimize import basinhopping, minimize, shgo
 
-from mesmer.mesmer_x.temporary_spatial import *
-from mesmer.mesmer_x.train_utils_mesmerx import *
+from mesmer.mesmer_x.temporary_spatial import (
+    gaspari_cohn_correlation_matrices,
+    geodist_exact,
+)
```
Comment (mathause) [inline mesmer/mesmer_x/train_l_distrib_mesmerx.py:21]:
```suggestion
from mesmer.geospatial import geodist_exact
from mesmer.stats import gaspari_cohn_correlation_matrices
```

File: mesmer/mesmer_x/train_l_distrib_mesmerx.py [hunk 3]
```diff
@@ -163,16 +170,18 @@ def xr_train_distrib(
...
+        lat_l_vec = predictors.lat  # lat["grid"][ls["idx_grid_l"]]
         # ... and function of MESMERv1
         geodist = geodist_exact(lon_l_vec, lat_l_vec)
 
         corr_gc = gaspari_cohn_correlation_matrices(geodist, [r_gasparicohn_2ndfit])[
             r_gasparicohn_2ndfit
         ]
         ind_nonan = np.where(
-            np.isnan(coefficients_xr[expression_fit.coefficients_list[0]]) == False
+            np.isnan(coefficients_xr[expression_fit.coefficients_list[0]]) is False
```
Comment (mathause) [inline mesmer/mesmer_x/train_l_distrib_mesmerx.py:184]:
```suggestion
            ~np.isnan(coefficients_xr[expression_fit.coefficients_list[0]])
```

**Review by mathause — COMMENTED** [review]
Just for my understanding - the idea is to merge this into #432 and then merge it into main?

**Comment by veni-vidi-vici-dormivi** [comment]
I want to exclude two files from the tests, I guess I should edit the pre commit config in another PR?

**Comment by mathause** [comment]
It currently fails because there is no `mesmer.geospatial` module (which is true, the stuff is in `mesmer.core.geospatial` so it needs `import mesmer` and we need to call `mesmer.spatial.calc_geodist_exact(...)`. Sorry about that. 

> I want to exclude two files from the tests, I guess I should edit the pre commit config in another PR?

You want to remove them from running via pytest? I think that should be possible here. But it might be best to get it to run locally, as these import conflicts can be quite annoying...

**Comment by veni-vidi-vici-dormivi** [comment]
> so it needs import mesmer and we need to call mesmer.spatial.calc_geodist_exact(...).

Why can't we use mesmer.core.geospatial.geodist_exact()`?

> But it might be best to get it to run locally, as these import conflicts can be quite annoying...

So it fails because OLD_train_l_distrib.py and `create_emus_l_dristrib.py` use functions from a mesmer.utils that does not exist anymore. I talked with Yann about this in #438, and it is because these two are old files which should be replaced by the other/new files. This is why I think it makes sense to exclude them from any testing.

**Comment by mathause** [comment]
I think the fix now is to move example_mesmer_x.py to the example folder (or to wrap the code in the file to `if __name__ == "__main__"` block, but the example code should not live in the mesmer code folder).

**Comment by mathause** [comment]
Love your commit names ;-) At the risk of making myself _very_ unpopular - I prefer the comments above the code (but ok to leave it like that if it's too stupid for you)

**Comment by veni-vidi-vici-dormivi** [comment]
>  I prefer the comments above the code (but ok to leave it like that if it's too stupid for you)

Well that actually makes two of us... I'm switching to a more entertaining Spotify playlist for that one...

**Comment by mathause** [comment]
Now you are just showing off 😉

**Comment by veni-vidi-vici-dormivi** [comment]
Okay this is ready imo.

**Comment by veni-vidi-vici-dormivi** [comment]
Is there no branch protection on this one? I dont see that I need a review...

**Comment by mathause** [comment]
Probably because you don't want to merge into main. I am looking at the code now.

File: mesmer/mesmer_x/train_utils_mesmerx.py [hunk 4]
```diff
@@ -339,28 +343,29 @@ def evaluate(self, coefficients_values, inputs_values, forced_shape=None):
...
         for c in coefficients_values:
-            exec(c + " = coefficients_values[c]")  # using the evil exec...
+            exec(c + " = coefficients_values[c]")
+            # using the evil exec...
 
         # Evaluation 2: inputs
         for i in inputs_values:
-            exec(i + " = inputs_values[i]")  # the evil exec strikes back...
+            exec(i + " = inputs_values[i]")
+            # the evil exec strikes back...
```
Comment (mathause) [inline mesmer/mesmer_x/train_utils_mesmerx.py:352]:
```suggestion
            exec(i + " = inputs_values[i]")
```

File: mesmer/mesmer_x/train_utils_mesmerx.py [hunk 5]
```diff
@@ -339,28 +343,29 @@ def evaluate(self, coefficients_values, inputs_values, forced_shape=None):
 
         # Evaluation 1: coefficients
         for c in coefficients_values:
-            exec(c + " = coefficients_values[c]")  # using the evil exec...
+            exec(c + " = coefficients_values[c]")
+            # using the evil exec...
```
[the remark this answers is not in the document]
Comment (mathause) [reply mesmer/mesmer_x/train_utils_mesmerx.py:347]: yes, would prefer not to use that...

File: mesmer/mesmer_x/train_utils_mesmerx.py [hunk 6]
```diff
@@ -339,28 +343,29 @@ def evaluate(self, coefficients_values, inputs_values, forced_shape=None):
...
 
         # Evaluation 3: parameters
         self.parameters_values = {}
         for param in self.parameters_list:
             # may need to silence warnings here, to avoid spamming
-            self.parameters_values[param] = eval(
-                self.parameters_expressions[param]
-            )  # the evil exec evolves as evil eval!
+            self.parameters_values[param] = eval(self.parameters_expressions[param])
+            # the evil exec evolves as evil eval!
```
Comment (mathause) [inline mesmer/mesmer_x/train_utils_mesmerx.py:359]:
```suggestion
            self.parameters_values[param] = eval(self.parameters_expressions[param])
```

File: mesmer/mesmer_x/__init__.py [hunk 7]
```diff
@@ -8,7 +8,6 @@
 
 # flake8: noqa
 # from .create_emus_l_distrib import *
```
Comment (mathause) [inline mesmer/mesmer_x/__init__.py:10]:
```suggestion
# TODO: replace * imports
# from .create_emus_l_distrib import *
```

File: mesmer/mesmer_x/create_emus_l_distrib.py
[the remark this answers is not in the document]
Comment (mathause) [reply mesmer/mesmer_x/create_emus_l_distrib.py:1]: If so please comment - otherwise maybe comment on the dummy functions

**Review by mathause — APPROVED** [review]
Some comments that I found by quickly going over the files and that should be in the spirit of your suggestions (i.e. formatting and variable names).

File: mesmer/mesmer_x/OLD_train_l_distrib.py [hunk 8]
```diff
@@ -987,21 +1020,21 @@ def reglin_fg(self, typ_cov, data):
...
                 for i, form in enumerate(self.cov["cov_" + typ_cov + "_form"])
                 if form in self.possible_sigmoid_forms
             ]
         else:
             inds_sigm = []
-        if (
-            len(inds_sigm) > 0
-        ):  # checking that there are covariates on this parameter with lositic form
+
+        # checking that there are covariates on this parameter with lositic form
```
[the remark this answers is not in the document]
Comment (veni-vidi-vici-dormivi) [reply mesmer/mesmer_x/OLD_train_l_distrib.py:1035]: I guess so

File: mesmer/mesmer_x/create_emus_l_distrib.py
[the remark this answers is not in the document]
Comment (veni-vidi-vici-dormivi) [reply mesmer/mesmer_x/create_emus_l_distrib.py:1]: Yes it is also an old one. Will do.

**Comment by veni-vidi-vici-dormivi** [comment]
Ah I like that a little better thanks!

**Comment by veni-vidi-vici-dormivi** [comment]
It's still failing, but they don't give any information as to why?

**Comment by veni-vidi-vici-dormivi** [comment]
🎉

**Pull request merged**